editor/scene_tree_dock.cpp

```diff
@@ -157,7 +158,11 @@ void SceneTreeDock::shortcut_input(const Ref<InputEvent> &p_event) {
 	}
 
 	if (ED_IS_SHORTCUT("scene_tree/rename", p_event)) {
-		_tool_selected(TOOL_RENAME);
+		// Prevent renaming if a button is focused
+		// to avoid conflict with Enter shortcut on macOS
+		if (!focus_owner || !Object::cast_to<BaseButton>(focus_owner)) {
```

Comment:
That's... very specific 🙃
The issue only exists because of a default shortcut. Shortcuts are rebindable, so it doesn't solve all problems.
But eh, technically it fixes the linked issue.

Reply:
It is very specific and the code is not as pretty :). But previous code didn't make sense though... Blocking all shortcuts seems excessive, even if it's a rare encounter it doesn't feel good if nothing happens when you press a shortcut. If user rebinds to conflicting shortcuts it's a different story I think